app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt

```diff
@@ -0,0 +1,162 @@
+import androidx.core.content.ContextCompat
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import kotlinx.android.synthetic.main.main_activity.bottomNav
+import kotlinx.android.synthetic.main.profile_view.*
+
+class ProfileActivity : AppCompatActivity() {
+
+    private val CAMERA_PERMISSION_REQUEST_CODE = 2000
+    private val CAMERA_REQUEST_CODE = 2001
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.profile_view)
+        val profileNameText = findViewById<TextView>(R.id.txtPersonName)
+        val emailAddressText = findViewById<TextView>(R.id.txtEmailAddress)
+
+        /** Changes the text views to previously set text*/
+        profileNameText.text = intent.extras?.getString("profileName")
+        emailAddressText.text = intent.extras?.getString("emailAddress")
+
+        /** Navigation Bar for bottom of screen*/
+        bottomNav.setOnNavigationItemSelectedListener {
+            when (it.itemId) {
+                R.id.search_menu -> {
+                    val intent = Intent(this, SearchActivity::class.java)
+                    startActivity(intent)
+                    true
+                }
+                R.id.library_menu -> {
+                    val intent = Intent(this, MainActivity::class.java)
+                    startActivity(intent)
+                    true
+                }
+                R.id.profile_menu -> {
+                    true
+                }
+                R.id.settings_menu -> {
+                    true
+                }
+                else -> false
+            }
+        }
+        /** Opens Dialog Screen to change profile name*/
+        val changeNameBtn = findViewById<Button>(R.id.btnChangeName)
+        changeNameBtn.setOnClickListener {
+            showChangeNameDialog()
+        }
+        /** Opens Dialog Screen to change email address */
+        val changeEmailBtn = findViewById<Button>(R.id.btnChangeEmail)
+        changeEmailBtn.setOnClickListener {
+            showChangeEmailDialog()
+        }
+        /** Opens the camera to take a picture for the profile photo*/
+        val photoBtn = findViewById<FloatingActionButton>(R.id.btnCamera)
+        photoBtn.setOnClickListener {
+            prepTakePhoto()
+        }
+    }
+
+    /** Check for permissions to use camera*/
+    private fun prepTakePhoto() {
+        if (ContextCompat.checkSelfPermission(
+                applicationContext!!,
+                Manifest.permission.CAMERA
+            ) == PackageManager.PERMISSION_GRANTED
+        ) {
+            takePhoto()
+        } else {
+            val permissionRequest = arrayOf(Manifest.permission.CAMERA)
+            requestPermissions(permissionRequest, CAMERA_PERMISSION_REQUEST_CODE)
+        }
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
```

Comment:
What does the `out` do here?

Reply:
Not sure, I was following the prof's tutorial. I removed it and it seems to work fine.